force-app/main/default/classes/StringGenerator.cls

```diff
@@ -0,0 +1,30 @@
+public with sharing class StringGenerator {
+
+    private List<String> my_list = new List<String>();
+
+    public StringGenerator( List<String> my_list_by_user )
+    {
```

Comment:
why do you want to pass the list of strings to the constructor instead of directly to the method being called? and why do you need to have a member list for this class?

Reply:
Should I keep it this way or should I remove it as a data member?